gamemaster/adjudicator/pregame.php

```diff
@@ -1,322 +1,231 @@
-		foreach($Game->Members->ByID as $Member)
-		{
-			$Member->countryID = $userCountries[$Member->userID];
-			$Game->Members->ByCountryID[$Member->countryID] = $Member;
-		}
-
-		for($countryID=1; $countryID<=count($Game->Variant->countries); $countryID++)
-			assert('$Game->Members->ByCountryID[$countryID]->countryID==$countryID');
-	}
-
-	protected function assignTerritories() {
-		global $DB, $Game;
-
-		$DB->sql_put(
-			"INSERT INTO wD_TerrStatus ( gameID, countryID, terrID )
-			SELECT ".$Game->id." as gameID, countryID, id
-			FROM wD_Territories
-			WHERE countryID > 0 AND mapID=".$Game->Variant->mapID." AND (coast='No' OR coast='Parent')"
-		);
-	}
-
-	protected function assignUnits() {
-		global $DB, $Game;
-
-		$terrIDByName = array();
-		$tabl = $DB->sql_tabl("SELECT id, name FROM wD_Territories WHERE mapID=".$Game->Variant->mapID);
-		while(list($id, $name) = $DB->tabl_row($tabl))
-			$terrIDByName[$name]=$id;
-
-		$UnitINSERTs = array();
-		foreach($this->countryUnits as $countryName => $params)
-		{
-			$countryID = $Game->Variant->countryID($countryName);
-
-			foreach($params as $terrName=>$unitType)
-			{
-				$terrID = $terrIDByName[$terrName];
-
-				$UnitINSERTs[] = "(".$Game->id.", ".$countryID.", '".$terrID."', '".$unitType."')"; // ( gameID, countryID, terrID, type )
-			}
-		}
-
-		$DB->sql_put(
-			"INSERT INTO wD_Units ( gameID, countryID, terrID, type )
-			VALUES ".implode(', ', $UnitINSERTs)
-		);
-	}
-
-	protected function assignUnitOccupations() {
-		global $DB, $Game;
-
-		// Now link the TerrStatus and Units records via the occupyingUnitID TerrStatus column
-		$DB->sql_put(
-			"UPDATE wD_TerrStatus t
-			INNER JOIN wD_Units u
-				ON (
-					t.gameID = u.gameID
-					/* TerrStatus does not deal with coasts */
-					AND ".$Game->Variant->deCoastCompare('t.terrID','u.terrID')."
-				)
-			SET t.occupyingUnitID = u.id
-			WHERE u.gameID = ".$Game->id
-		);
-	}
-
-	/**
-	 * Initialize the game (more of a phase change than adjudication). Will throw an exception
-	 * if the game doesn't have enough players, which will be caught in gamemaster.php and result
-	 * in the game's deletion
-	 *
-	 * Deletes game and throws exception if game cannot start
-	 */
-	function adjudicate()
-	{
-		global $Game;
-
-		// Will give back bets, send messages, delete the game, and throw an exception to get back to gamemaster.php
-		if( !$this->isEnoughPlayers() ) $Game->setNotEnoughPlayers();
-
-
-		// Determine which countryID is given to which userID
-		$userCountries = $this->userCountries();// $userCountries[$userID]=$countryID
-
-		assert('count($userCountries) == count($Game->Variant->countries) && count($userCountries) == count($Game->Members->ByID)');
-
-		$this->assignCountries($userCountries);
-
-		// Create starting board conditions, typically based on $countryUnits
-		$this->assignTerritories(); // TerrStatus
-		$this->assignUnits(); // Units
-		$this->assignUnitOccupations(); // TerrStatus occupyingUnitID
-	}
-}
-
+<?php
```

Comment:
The permissions on this file have changed - we should probably leave them as is, unless there's a good reason to change.

Reply:
Will be in next commit.

Reply:
Is that commit coming? I'd really like to accept this PR :)